tests/unit/readers/test_reader_postgres.py

```
@@ -360,32 +327,122 @@ class TestCheckHealth:
 
     def test_unhealthy_when_load_raises_runtime_error(self, reader: ReaderPostgres) -> None:
-        """Test returns unhealthy when _load_db_config raises RuntimeError."""
-        with patch.object(reader, "_load_db_config", side_effect=RuntimeError("Failed to load.")):
-            healthy, message = reader.check_health()
+        """Test raises HealthCheckError when _pg_config raises RuntimeError."""
+        with patch.object(
+            type(reader),
+            "_pg_config",
+            new_callable=lambda: property(lambda self: (_ for _ in ()).throw(RuntimeError("Failed to load."))),
+        ):
+            with pytest.raises(HealthCheckError, match="Failed to load."):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)